### DISCUSSION TITLE
During a test interpret enter as space

### DISCUSSION THREAD

**Discussion by SethFalco** [monkeytypegame/monkeytype#2225] [Ideas]
To learn touch typing, I'm relying on 2 projects:
* [KTouch](https://apps.kde.org/ktouch/) by KDE
* Monkeytype

A habit that I've gotten from KTouch is that it uses spaces between words, as normal, but at the end of the line we have to press `ENTER` by default to move to the next one.

This can be a pain since on Monkeytype, I often get to the end of a line then press enter instead of space since it has the appearance of the next line. (Which doesn't do anything at the moment.) This isn't a big deal, and the behavior in KTouch can be configured to match Monkeytype, though.

I was wondering, could it make sense to interpret the `ENTER` key as a space during a typing test?

**Comment by Miodec** [comment]
I could add this for texts that do not have a new line character, but then it could be a bit confusing, where sometimes you can press enter whenever and sometimes you can only press enter in certain situations.

Or maybe allow enter and space always, unless its a new line character then only allow enter?

**Reply by SethFalco** [reply]
If it's going to cause an inconsistent experience, then it's probably not worth doing tbh.
Sorry, I didn't realize tests could have text with line breaks in it.

Depending on the length of the text, it might not be immediately apparent that it contains line breaks too. (Not sure, I haven't encountered a test with line breaks.) I would imagine some users would have to wonder what the behavior of `ENTER` would be until they try, or otherwise see a line break later in the text.

It might be that this was a bad idea then.

Edit: Missed the 2nd part of your comment somehow…

> Or maybe allow enter and space always, unless its a new line character then only allow enter?

That could be a good solution, but it might also feel weird. If the text contains actual paragraphs/line breaks, it'd feel wrong to allow the use of `ENTER`. It would technically resolve the qualm I had, though.

Now that I know the text can have line breaks, I'd probably withdraw my initial suggestion, though. Unless you think it's worth doing anyway.
